src/carbon_calculator/carbonCalculator.py

```diff
@@ -43,20 +46,32 @@ def SavePic2Media(picURL):
             print("ERROR: Unable to import action photo from "+picURL)
             return None
         else:
-            file_name =  "carbon_calculator/assets/" + picURL.split("/")[-1]  # There's probably a better way of doing this but this is just a quick example
-            fp = open(file_name,"wb")
-            fp.write(resp.content)
-            fp.close()
-            media = CarbonCalculatorMedia(file = file_name)
-            #media = CarbonCalculatorMedia.objects.create(file=file_name)
+            image = resp.content
+            file_name =  picURL.split("/")[-1]
+            file_type_ext = file_name.split(".")[-1]
+
+            img_io = BytesIO(image)
+            
+            content_type = 'image/jpeg'
+            if file_type_ext.lower() == 'png':
+                content_type = 'image/png'
+
+            # Create a new Django file-like object to be used in models as ImageField using
+            # InMemoryUploadedFile.  If you look at the source in Django, a
+            # SimpleUploadedFile is essentially instantiated similarly to what is shown here
+            #image_file = InMemoryUploadedFile(img_io, None, 'foo.jpg', 'image/jpeg',
+            image_file = InMemoryUploadedFile(img_io, None, file_name, content_type,
+                                  None, None)
+
+            media = CarbonCalculatorMedia.objects.create(file=image_file, name=f"{slugify(file_name)}")
```

Comment:
i am a bit confused.  i think you can just pass in image here and it will be fine.  At this point it should already be in memory so no need to redeclare an InMememoryUploadedFile

Reply:
There may be other ways to do it, but I tried a lot of things that didn't work including passing the image in directly.  This came from an example I found, and works for both png and jpg.  it was important to specify the content_type for the in memory file.